spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/EntityRowMapper.java

```diff
@@ -69,155 +73,170 @@ public EntityRowMapper(RelationalPersistentEntity<T> entity, RelationalMappingCo
 	 */
 	@Override
 	public T mapRow(ResultSet resultSet, int rowNumber) {
+		return new ReadingContext(resultSet).mapRow();
+	}
 
-		String prefix = "";
+	@Value
+	@RequiredArgsConstructor
+	private class ReadingContext {
+		ResultSet resultSet;
+		PersistentPropertyPathExtension path;
 
-		RelationalPersistentProperty idProperty = entity.getIdProperty();
+		ReadingContext(ResultSet resultSet) {
+			this.resultSet = resultSet;
+			this.path = new PersistentPropertyPathExtension(context, entity);
+		}
 
-		Object idValue = null;
-		if (idProperty != null) {
-			idValue = readFrom(resultSet, idProperty, prefix);
+		private ReadingContext extendBy(RelationalPersistentProperty property) {
+			return new ReadingContext(resultSet, path.extendBy(property));
 		}
 
-		T result = createInstance(entity, resultSet, idValue, prefix);
+		T mapRow() {
 
-		return entity.requiresPropertyPopulation() //
-				? populateProperties(result, resultSet) //
-				: result;
-	}
+			RelationalPersistentProperty idProperty = entity.getIdProperty();
```

Comment:
It would make sense to move all this mapping to a `MappingR2dbcConverter` to keep `EntityRowMapper` slim (see https://github.com/spring-projects/spring-data-r2dbc/blob/master/src/main/java/org/springframework/data/r2dbc/function/convert/EntityRowMapper.java#L29). Doing so would align object mapping with the other Spring Data modules.